File: ImasLiveDB-Android/app/src/main/kotlin/com/fugaif/imaslivedb/ui/components/GradientHeader.kt
package com.fugaif.imaslivedb.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Gradient header background for idol detail screens.
 * Fades from the idol's image color (60% opacity, top-left) to near-transparent (bottom-right).
 * Mirrors iOS GradientHeader.
 */
@Composable
fun GradientHeader(
    color: Color,
    height: Dp = 120.dp,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .height(height)
            .background(
                brush = Brush.linearGradient(
                    colors = listOf(
                        color.copy(alpha = 0.6f),
                        color.copy(alpha = 0.1f)
                    )
                )
            )
    )
}

File: ImasLiveDB/Views/Components/GradientHeader.swift
import SwiftUI

/// アイドル詳細ヘッダ — イメージカラーのグラデーション背景
struct GradientHeader: View {
    let color: Color

    var body: some View {
        LinearGradient(
            colors: [color.opacity(0.6), color.opacity(0.1)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
        .frame(height: 120)
    }
}
